tests/ImageSharp.Tests/Image/ImageTests.Load_FromStream_PassLocalConfiguration.cs

```diff
@@ -39,14 +39,25 @@ public void Configuration_Stream_Agnostic()
             [Fact]
             public void NonSeekableStream()
             {
-                var stream = new NoneSeekableStream(this.DataStream);
+                var stream = new NonSeekableStream(this.DataStream);
```

Comment:
These tests are insufficient.  They are only here to prove the different overloads of extension methods do what they promise by mocking a decoder. `this.DataStream` contains only 16 bytes!

At this point, I would add a couple of integration tests, stressing all decoders with `NonSeekableStream`, and images of different sizes.

Reply:
I've copied the existing `Image.Identify` and `Image.IdentifyAsync` to demonstrate accuracy. The unit tests also cover reading and writing across chunk boundaries.